index.js

```diff
@@ -1,36 +1,75 @@
 'use strict';
-module.exports = leftPad;
 
-var cache = [
-  '',
-  ' ',
-  '  ',
-  '   ',
-  '    ',
-  '     ',
-  '      ',
-  '       ',
-  '        ',
-  '         '
+module.exports  = leftPad;
+
+leftPad.spaces  = [ // avoid global pollution & power of 2 size
+    ''
+  , ' '                 //1
+  , '  '                //2
+  , '   '           
+  , '    '              //4
+  , '     '         
+  , '      '
+  , '       '
+  , '        '          //8
+  , '         '
+  , '          '
+  , '           '
+  , '            '
+  , '             '
+  , '              '
+  , '               '
+  , '                '  //16
 ];
 
-function leftPad (str, len, ch) {
-  // convert `str` to `string`
-  str = str + '';
+leftPad.zeros   = [ // that's another most used case, worth to cache it
+    ''
+  , '0'                 //1
+  , '00'                //2
+  , '000'           
+  , '0000'              //4
+  , '00000'         
+  , '000000'
+  , '0000000'
+  , '00000000'          //8
+  , '000000000'
+  , '0000000000'
+  , '00000000000'
+  , '000000000000'
+  , '0000000000000'
+  , '00000000000000'
+  , '000000000000000'
+  , '0000000000000000'  //16
+];
 
-  // doesn't need to pad
+function leftPad (str, len, ch) {
+  var   pad     = ''
```

Comment:
@cloned2k16 did you mean to write the var definition like this?
`var pad = '', cache;`